Client/weather-app/src/components/Weather.vue

```diff
@@ -0,0 +1,471 @@
+            <div class="weatherr-history-container">
+                    <div v-for="(weather, index) in weatherHistory" :key="weather.date" class="flex">
+                        <div v-for="(w) in weather" :key="w.date">
+                            <div class="weather">
+                                <div class="current">
+                                <div class="info">
+                                    <div>&nbsp;</div>
+                                    <div class="city">
+                                    <small>
+                                        <small>CITY:</small>
+                                    </small>
+                                    {{ w.city }}
+                                    </div>
+                                    <div class="temp">
+                                    <small>
+                                        <small>TEMP:</small>
+                                    </small>
+                                    {{w.temperature}}&deg;
+                                    <small>C</small>
+                                    </div>
+                                    <div class="wind">
+                                    <small>
+                                        <small>WIND:</small>
+                                    </small>
+                                    {{w.windSpeed}} km/h
+                                    </div>
+                                    <div class="humidity">
+                                    <small>
+                                        <small>HUMIDITY:</small>
+                                    </small>
+                                    {{w.humidity}}%
+                                    </div>
+                                    <div>&nbsp;</div>
+                                </div>
+                                <div class="icon">
+                                    <!-- <div class="sunny"></div> -->
+                                     <div :class=w.weatherDescription></div>
+                                </div>
+                                <div class="future">
+                                <div class="day" :class="{ 'current-weather': index === 0 }">
+                                    <h4 class="date">{{w.date}}</h4>
+                                </div>
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+            </div>
+            </div>     
+</div>
+ 
+</template>
+
+<script>
+import axios from "axios";
+
+export default {
+    name: 'Weather',
+    
+    data () {
+        return {
+            weatherForecast: [],
+            inputValue: 'Bonn',
+            weatherHistory: [],
+            showError: false,
+            overcastCloudy: 'snowy'
+        }
+    },
+
+    computed: {
+        axiosParams() {
+            const params = new URLSearchParams();
+            if (typeof(this.inputValue) === 'string') {
```

Comment:
Checking for the type of an input is not reliable mechanism to determine which field it is. What happens if you add a second string field - Your app will break. You should find a mechanism to determine the name of the field